### ISSUE TITLE
Possibly undefined but it's not

### ISSUE THREAD

**Issue by mbayou** [Microsoft/TypeScript#19415]
Hi,

I have an issue about the `strictNullChecks`, indeed I am using it right now and it says "Object is possibly 'undefined'." on an object cannot be undefined.  

**TypeScript Version:** 2.5.3

**Code**

Failling code

```ts
// Retrieve circle with identifier
const circle: Circle | undefined = await this.circleDao.findById(command.getId());
// Check if localAuthority is found in database
Utils.checkArgument(circle !== undefined, "Circle with id '" + command.getId() + "' cannot be found");

// Compiler fail on this line with message "Object is possibly 'undefined'."
circle.setDefaultCircle(command.isDefaultCircle());
```

Utils.ts method
```ts
public static checkArgument(expression: boolean, message?: string) {
        if (!expression) {
            throw new IllegalArgumentError(message);
        }
}
```

**Expected behavior:**

When I check if an object is undefined in an external method the compiler can see it and don't throw error : "Object is possibly 'undefined'."

**Actual behavior:**

When I check if an object is undefined in an external method the compiler throw error : "Object is possibly 'undefined'."

**Comment by Ghabriel** [comment]
See #9998 .

**Comment by mhegazy** [comment]
also see https://github.com/Microsoft/TypeScript/issues/16546 and https://github.com/Microsoft/TypeScript/issues/9655

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
